### PR TITLE
chore: deploy from actions

### PR DESCRIPTION
**Pull request by redreceipt**

### CODE REVIEW

File: scripts/setup-app.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
+if [ -z "$CHURCH" ] || [ -z "$APOLLOS_API_KEY" ]; then
+    echo "CHURCH and APOLLOS_API_KEY required!"
+    exit 1
+fi
+
+API_URL="https://apollos-cluster-production.herokuapp.com/api"
+APP=$(curl -s --fail -H "x-api-key: $APOLLOS_API_KEY" -H "accept: text/plain" "$API_URL/config/$CHURCH/APP_NAME")
+IOS_BUNDLE=$(curl -s --fail -H "x-api-key: $APOLLOS_API_KEY" -H "accept: text/plain" "$API_URL/config/$CHURCH/APP.APPLE_BUNDLE_ID")
+ANDROID_BUNDLE=$(curl -s --fail -H "x-api-key: $APOLLOS_API_KEY" -H "accept: text/plain" "$API_URL/config/$CHURCH/APP_ANDROID_PKG_ID")
```
Comment (vinnyjth) [inline scripts/setup-app.sh:9]:
```suggestion
ANDROID_BUNDLE=$(curl -s --fail -H "x-api-key: $APOLLOS_API_KEY" -H "accept: text/plain" "$API_URL/config/$CHURCH/APP.ANDROID_PKG_ID")
```

File: scripts/setup-app.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+SERVER_URL=$(curl -s --fail -H "x-api-key: $APOLLOS_API_KEY" -H "accept: text/plain" "$API_URL/config/$CHURCH/APP_SERVER_URL")
```
Comment (vinnyjth) [inline scripts/setup-app.sh:10]:
```suggestion
SERVER_URL=$(curl -s --fail -H "x-api-key: $APOLLOS_API_KEY" -H "accept: text/plain" "$API_URL/config/$CHURCH/APP.SERVER_URL")
```

maybe in the future we could come up with a way to pull this from the church_routes table :)

Comment (redreceipt) [reply scripts/setup-app.sh:10]: Yea I wanted to. but the fact that we have a routes table prevents that :)

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -22,17 +17,16 @@ jobs:
...
-      - run: node ./scripts/get-introspection-data.js
-      - run: apollos mobile deploy ios publish
+      - run: scripts/setup-app.sh
+        env:
+          APOLLOS_API_KEY: ${{ secrets.APOLLOS_API_KEY }}
+          CHURCH: apollos_demo
+      - run: (cd deploys/mobile/apollos_demo) && scripts/deploy-ios.sh
+        env:
+          APOLLOS_API_KEY: ${{ secrets.APOLLOS_API_KEY }}
+          CHURCH: apollos_demo
```
Comment (vinnyjth) [inline .github/workflows/deploy.yml:27]:
```suggestion
          CHURCH: ${{ github.event.inputs.church }}
```

Comment (vinnyjth) [reply .github/workflows/deploy.yml:27]: https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/

Comment (redreceipt) [reply .github/workflows/deploy.yml:27]: Nice! Thanks :) I was gonna figure that out after fixing whatever doesn't work post merge 😂

File: scripts/deploy-ios.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+if [ -n "$(git status --porcelain)" ]; then
+  echo "Git status not clean!"
+  git status --porcelain
+  exit 1
+fi
+
+API_URL="https://apollos-cluster-production.herokuapp.com/api"
+MATCH_GIT_BASIC_AUTHORIZATION=$(curl -s --fail -H "accept: text/plain" -H "x-api-key: $APOLLOS_API_KEY" "$API_URL/config/$CHURCH/APP_MATCH_TOKEN")
+MATCH_PASSWORD=$(curl -s --fail -H "accept: text/plain" -H "x-api-key: $APOLLOS_API_KEY" "$API_URL/config/$CHURCH/APP_MATCH_PASSWORD")
```
Comment (vinnyjth) [inline scripts/deploy-ios.sh:16]:
```suggestion
MATCH_PASSWORD=$(curl -s --fail -H "accept: text/plain" -H "x-api-key: $APOLLOS_API_KEY" "$API_URL/config/$CHURCH/APP.MATCH_PASSWORD")
```

File: scripts/deploy-ios.sh [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+fi
+
+if [ -n "$(git status --porcelain)" ]; then
+  echo "Git status not clean!"
+  git status --porcelain
+  exit 1
+fi
+
+API_URL="https://apollos-cluster-production.herokuapp.com/api"
+MATCH_GIT_BASIC_AUTHORIZATION=$(curl -s --fail -H "accept: text/plain" -H "x-api-key: $APOLLOS_API_KEY" "$API_URL/config/$CHURCH/APP_MATCH_TOKEN")
```
Comment (vinnyjth) [inline scripts/deploy-ios.sh:15]:
```suggestion
MATCH_GIT_BASIC_AUTHORIZATION=$(curl -s --fail -H "accept: text/plain" -H "x-api-key: $APOLLOS_API_KEY" "$API_URL/config/$CHURCH/APP.MATCH_TOKEN")
```

File: scripts/deploy-ios.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+MATCH_PASSWORD=$(curl -s --fail -H "accept: text/plain" -H "x-api-key: $APOLLOS_API_KEY" "$API_URL/config/$CHURCH/APP_MATCH_PASSWORD")
+MATCH_GIT_URL=$(curl -s --fail -H "accept: text/plain" -H "x-api-key: $APOLLOS_API_KEY" "$API_URL/config/$CHURCH/APP_MATCH_REPO")
```
Comment (vinnyjth) [inline scripts/deploy-ios.sh:17]:
```suggestion
MATCH_GIT_URL=$(curl -s --fail -H "accept: text/plain" -H "x-api-key: $APOLLOS_API_KEY" "$API_URL/config/$CHURCH/APP.MATCH_REPO")
```

**Review by vinnyjth — APPROVED** [review]

**Comment by vinnyjth** [comment]
Cycle finished ;)

**Pull request merged**